horace_euphonic_interface/euphonic_wrapper.py

```diff
@@ -0,0 +1,247 @@
+        If non-zero will chunk the phonon calculation into blocks of <chunk> q-points
+    lim : float (default: infinity)
+        A cut-off for the calculated structure factor where values above `lim` is set equal to it.
+    scattering_lengths : dict or string (default: 'Sears1992')
+        The scattering lengths (in femtometres) to use for the phonon calculations either as 
+        a dictionary with elements as keys or a string denoting a database (internal or from file)
+    asr : {'realspace', 'reciprocal'}, optional
+    dipole : boolean, optional
+    eta_scale : float, optional
+    splitting : boolean, optional
+    insert_gamma : boolean, optional
+    reduce_qpts : boolean, optional
+    use_c : boolean, optional
+    n_threads : int, optional
+    fall_back_on_python : boolean, optional
+        These are parameters used in the `calculate_qpoint_phonon_modes` method of ForceConstants
+        Type `help(fc.calculate_qpoint_phonon_modes)` where fc is the ForceConstants object you created.
+    """
+    # This a wrapper around the Euphonic ForceConstants and QpointPhononModes classes to make it easier to access from Matlab
+    # It is meant to be used with a Matlab python_wrapper class and implements a horace_sqw function for use with Horace
+
+    defaults = {'debye_waller': None, 'debye_waller_grid': None, 'temperature': 0.0 * ureg('K'), 'bose': True,
+                'negative_e': False, 'conversion_mat': None, 'chunk': 5000, 'lim': np.inf, 'scattering_lengths': 'Sears1992',
+                'weights': None, 'asr': None, 'dipole': True, 'eta_scale': 1.0, 'splitting': True, 'insert_gamma': False, 
+                'reduce_qpts': True, 'use_c': False, 'n_threads': 1, 'fall_back_on_python': True}
+
+    def __init__(self, force_constants, **kwargs):
+        for key, val in self.defaults.items():
+            setattr(self, key, kwargs.pop(key, self.defaults[key]))
+        self.force_constants = force_constants
+
+    def _calculate_sf(self, qpts):
+        if self.temperature > 0:
+            if self.debye_waller is None and self.debye_waller_grid is not None:
+                self._calculate_debye_waller()
+        phonons = self._calculate_phonon_modes(qpts)
+        sf_obj = phonons.calculate_structure_factor(scattering_lengths=self.scattering_lengths,
+                                                    dw=self.debye_waller)
+        w = sf_obj.frequencies.magnitude
+        sf = sf_obj.structure_factors.magnitude
+        if self.temperature > 0 and self.bose:
+            bose = sf_obj._bose_factor(self.temperature)
+            sf = (1 + bose) * sf
+        if self.negative_e:
+            w = np.hstack((w, -w))
+            neg_sf = sf_obj.structure_factors.magnitude
+            if self.temperature > 0 and self.bose:
+                neg_sf = bose * neg_sf
+            sf = np.hstack((sf, neg_sf))
+        return w, sf
+        
+    def horace_disp(self, qh, qk, ql, pars=[], *args, **kwargs):
```

Comment:
Only `horace_disp` is implemented here, but in the PR text `horace_sqw` is shown. It's fine if `horace_sqw` hasn't been implemented yet, but the PR example should probably be updated.